File: top-interview-questions-easy/Others/valid-parentheses/20.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  let stack = []
  s.split('').forEach((item) => {
    if (!stack.length) {
      stack.push(item)
      return
    }
    let top = stack[stack.length - 1]
    let dis = item.charCodeAt() - top.charCodeAt()
    if (dis === 1 || dis === 2 ) {
      stack.pop()
    } else {
      stack.push(item)
    }
  })
  return !stack.length
}

File: top-interview-questions-easy/Others/valid-parentheses/20.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        for i in s:
            if not len(stack):
                stack.append(i)
                continue
            top = stack[len(stack) - 1]
            dis = ord(i) - ord(top)
            if dis == 1 or dis == 2:
                stack.pop()
            else:
                stack.append(i)

        return not len(stack)
